src/routines.ts

```diff
@@ -495,6 +597,67 @@ export const verifyData = async function(
   return sign.detached.verify(payloadHash, theirPayloadSignature, theirPublicSignKey)
 };
 
+export const createPQSharedSecret = async function(
+  to: IIdentity
+): Promise<IPQSharedSecret> {
+
+  const { cipherText, sharedSecret } = ml_kem768.encapsulate(base64.decode(to.key.public.box));
```

Comment:
Need to call the correct kyber variant based on the key type selected in the `to` field